poetry/repositories/pypi_repository.py

```diff
@@ -221,9 +221,17 @@ def search(self, query, mode=0):
         hits = client.search(search, "or")
 
         for hit in hits:
-            result = Package(hit["name"], hit["version"], hit["version"])
-            result.description = to_str(hit["summary"])
-            results.append(result)
+            try:
+                result = Package(hit["name"], hit["version"], hit["version"])
+                result.description = to_str(hit["summary"])
+                results.append(result)
+            except ParseVersionError:
+                self._log(
+                    'Unable to parse version "{}" for the {} package, skipping'.format(
+                        hit["version"], hit["name"]
+                    ),
+                    level="debug",
```

Comment:
I wonder if this should be a warning that's printed to the shell? This way it much more likely that someone will raise the issue with the upstream.

Reply:
I think there is a fairly large number of malformed packages on pypi that are easy to run into when you search for a fragment of a package name. This is almost the same logging that is done in `find_packages` in the same file.

Reply:
Yep